src/core/public/application/scoped_history.test.ts

```diff
@@ -268,8 +268,18 @@ describe('ScopedHistory', () => {
       const gh = createMemoryHistory();
       gh.push('/app/wow');
       const h = new ScopedHistory(gh, '/app/wow');
-      expect(h.createHref({ pathname: '' })).toEqual(`/`);
+      expect(h.createHref({ pathname: '' })).toEqual(`/app/wow`);
       expect(h.createHref({ pathname: '/new-page', search: '?alpha=true' })).toEqual(
+        `/app/wow/new-page?alpha=true`
+      );
```

Comment:
I started working on https://github.com/elastic/kibana/issues/58137 and used this pr. 
Looks like in real plugin the base name has trailing slash. 
so for this test it would be instead of 
```
const h = new ScopedHistory(gh, '/app/wow');
```
it would be:
```
const h = new ScopedHistory(gh, '/app/wow/');
```

And then this test would fail with: `/app/wow//new-page?alpha=true` (2 slashes)

Reply:
You are right. This was caused by

```ts
  private prependBasePathToString(path: string): string {
    path = path.startsWith('/') ? path.slice(1) : path;
    return path.length ? `${this.basePath}/${path}` : this.basePath;
  }
```

Where `${this.basePath}/${path}` always inserts a `/` between the paths.

Fixed  in https://github.com/elastic/kibana/pull/62407/commits/460abe1b321826e09bdf08eaa181b4cf0d0b3909